File: 3090. Maximum Length Substring With Two Occurrences/maximumLengthSubstring.py
import sys
from collections import defaultdict

class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        left = 0
        right = 0
        result = 0
        map = defaultdict(int)
        while right < len(s):
            map[s[right]] += 1
            while (map[s[right]] > 2):
                map[s[left]] -= 1
                left += 1
            result = max(result, right - left + 1)
            right += 1
        return result

File: 3090. Maximum Length Substring With Two Occurrences/maximumLengthSubstring.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int left = 0;
        int right = 0;
        int max = 0;
        HashMap<Character, Integer> map = new HashMap<>();
        while (right < s.length()) {
            map.put(s.charAt(right), map.getOrDefault(s.charAt(right), 0) + 1);
            while (left <= right && map.get(s.charAt(right)) > 2) {
                int frequency = map.get(s.charAt(left));
                map.put(s.charAt(left), frequency - 1);
                left++;
            }
            max = Math.max(max, right - left + 1);
            right++;
        }
        return max;
    }
}